stl/inc/xlocnum

```diff
@@ -169,17 +158,22 @@ protected:
         _Truename  = nullptr;
 
         _Tidy_guard<numpunct> _Guard{this};
-        _Grouping      = _Maklocstr(_Isdef ? "" : _Ptr->grouping, static_cast<char*>(nullptr), _Lobj._Getcvt());
+        _Grouping      = _Maklocstr(_Isdef ? "" : _Ptr->grouping, static_cast<char*>(nullptr), _Cvt);
         _Falsename     = _Maklocstr(_Lobj._Getfalse(), static_cast<_Elem*>(nullptr), _Cvt);
         _Truename      = _Maklocstr(_Lobj._Gettrue(), static_cast<_Elem*>(nullptr), _Cvt);
         _Guard._Target = nullptr;
 
         if (_Isdef) { // apply defaults for required facets
-            // _Grouping = _Maklocstr("", static_cast<char *>(nullptr), _Cvt);
             _Dp         = _Maklocchr('.', static_cast<_Elem*>(nullptr), _Cvt);
             _Kseparator = _Maklocchr(',', static_cast<_Elem*>(nullptr), _Cvt);
         } else {
-            _Getvals(_Elem{}, _Ptr, _Cvt);
+            if constexpr (is_same_v<_Elem, wchar_t>) {
+                _Dp         = static_cast<_Elem>(_Ptr->_W_decimal_point[0]);
+                _Kseparator = static_cast<_Elem>(_Ptr->_W_thousands_sep[0]);

```

Comment:
As in `<xlocmon>`, we now know that `_Elem` is `wchar_t` here, so `_W_decimal_point` and `_W_thousands_sep` have the correct type and we should drop `static_cast<_Elem>`.